tests/testProfileChangePassword.php

```diff
@@ -81,58 +81,135 @@ public function testInitHooks() {
+		// Same method hooked to all 3 actions.
+		$class_name = 'WP_Auth0_Profile_Change_Password';
+		$this->assertHooked( 'user_profile_update_errors', $class_name, $expect_hooked );
+		$this->assertHooked( 'validate_password_reset', $class_name, $expect_hooked );
+		$this->assertHooked( 'woocommerce_save_account_details_errors', $class_name, $expect_hooked );
 	}
 
 	/**
-	 * Test that the validate_new_password method works as expected.
+	 * Test that empty password fields will skip password update.
 	 */
-	public function testValidateNewPassword() {
-		$errors = new WP_Error();
+	public function testThatEmptyPasswordFieldSkipsUpdate() {
+		$user_id = $this->createUser()->ID;
+		$errors  = new WP_Error();
 
-		$user_data = $this->createUser();
-		$user_id   = $user_data->ID;
-		$user_obj  = get_user_by( 'id', $user_id );
+		$mock_api_test_password = $this->getStub( true );
+		$change_password        = new WP_Auth0_Profile_Change_Password( $mock_api_test_password );
 
-		// Create a stub for the WP_Auth0_Api_Change_Password class.
-		$mock_api_test_password = $this
-			->getMockBuilder( WP_Auth0_Api_Change_Password::class )
-			->setMethods( [ 'call' ] )
-			->setConstructorArgs( [ self::$options, self::$api_client_creds ] )
-			->getMock();
-		$mock_api_test_password->method( 'call' )->willReturn( true, false );
-
-		$change_password = new WP_Auth0_Profile_Change_Password( $mock_api_test_password );
+		$_POST['pass1']   = uniqid();
+		$_POST['user_id'] = $user_id;
+		$this->storeAuth0Data( $user_id );
+		$this->assertTrue( $change_password->validate_new_password( $errors, false ) );
+		$this->assertEmpty( $errors->get_error_messages() );
 
-		// Call should fail because of a missing password.
+		// Test core WP password field.
+		unset( $_POST['pass1'] );
 		$this->assertFalse( $change_password->validate_new_password( $errors, false ) );
 
-		$_POST['pass1'] = uniqid();
+		$_POST['password_1'] = uniqid();
+		$this->assertTrue( $change_password->validate_new_password( $errors, false ) );
 
-		// Call should fail with a password because of a missing user ID.
+		// Test WooCommerce password field.
+		unset( $_POST['password_1'] );
 		$this->assertFalse( $change_password->validate_new_password( $errors, false ) );
+	}
+
+	/**
+	 * Test that empty user data will skip the password update.
+	 */
+	public function testThatMissingUserDataSkipsUpdate() {
+		$user_obj = $this->createUser();
+		$user_id  = $user_obj->ID;
+		$errors   = new WP_Error();
+
+		$mock_api_test_password = $this->getStub( true );
+		$change_password        = new WP_Auth0_Profile_Change_Password( $mock_api_test_password );
 
-		// Call should fail with a user object or user_id in $_POST because of no Auth0 data stored.
-		$this->assertFalse( $change_password->validate_new_password( $errors, $user_obj ) );
+		$_POST['pass1']   = uniqid();
 		$_POST['user_id'] = $user_id;
+		$this->storeAuth0Data( $user_id );
```

Comment:
> testThatMissingUserDataSkipsUpdate

this is not empty, or what do you mean by "missing user data"?

Reply:
Either missing form field with a user ID or missing when passed in by WP.